### ISSUE TITLE
Add BatteryStatus

### ISSUE THREAD

**Issue by Uzlopak** [libuv/libuv#4172]
Well. How about providing the battery status, as described in mdn for navigator.getBattery() 
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/getBattery

Win:
https://learn.microsoft.com/en-us/windows/win32/power/battery-information

Mac:
https://developer.apple.com/documentation/iokit/iopowersources_h#topics

Linux:
Read the status from /sys/class/power_supply/

**Comment by juanarbol** [comment]
I can work on the macOS part :)

**Comment by juanarbol** [comment]
I can work on the linux for arm64, as well but I need a bit more time for it :)

**Comment by oluan** [comment]
I can work on Windows implementation soon

@juanarbol To ensure we're aligned, could you specify the features you're working on?

I've seen your PR and I'll use it to start my branch.

Specifically, are you planning to implement the `BatteryManager` browser interface as outlined here:

https://developer.mozilla.org/en-US/docs/Web/API/BatteryManager ?

**Comment by juanarbol** [comment]
Yes, once we have everything landed in libuv, I can implement the battery thing in Node.js

**Comment by juanarbol** [comment]
Are we sure about Linux? I tried reading from the `power_supply` folder, empty from a Raspberry Pi and from a AWS machine as well:

```
ubuntu@ip-172-31-11-139:~$ cat /sys/class/power_supply/
cat: /sys/class/power_supply/: Is a directory
ubuntu@ip-172-31-11-139:~$ cat /sys/class/power_supply/
cat: /sys/class/power_supply/: Is a directory
ubuntu@ip-172-31-11-139:~$ ls /sys/class/power_supply/
ubuntu@ip-172-31-11-139:~$ uname -a
Linux ip-172-31-11-139 6.2.0-1017-aws #17~22.04.1-Ubuntu SMP Fri Nov 17 21:07:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by Uzlopak** [comment]
```sh
aras@aras-Lenovo-Legion-5-17ARH05H:~$ cat /sys/class/power_supply/BAT0/capacity
100

```

**Comment by juanarbol** [comment]
That works on laptops; for servers, it is empty; the same happens with macOS, I will propose returning a `UV_EPERM` when no resources available
